### ISSUE TITLE
Function lock on G14 2021

### ISSUE THREAD

**Issue by Aurel100** [seerge/g-helper#112]
Hello,
Great tool ! congrats !

I have a request that would be very helpful for all users (I think) : 
I rarely use F1 to F12, I don't want to push the "fn" button to change the screen brightness for example.
Is there any way to use "fn" by default instead of F1, F,2 etc ? 

a bit like "verr num" ?

Thank you

**Comment by Aurel100** [comment]
> There must be in bios option for something like that to happen. Assigning an FN key lock can cause the whole system to go crazy. So I don't think it will work properly.
Hey :)
There isn't, i've already searched. That's why I am now leaning towards Seerge for help ;)

**Comment by seerge** [comment]
@Aurel100 i can look into that, but that means app will have to catch all keystrokes (in order to make regular F1-F12 work).

**Comment by seerge** [comment]
I don't think that other models (at least among G14/G15) have native fn lock. In general it's very tricky feature, as currently windows intercepts many fn+fx hits and i'm not sure if it will be even possible (technically) to catch them anyhow

**Comment by lucidnx** [comment]
You can try with AutoHotkey software. it should work, otherwise there's no way to do that.

**Comment by Aurel100** [comment]
> AutoHotkey

since your answer, i've taken interest in programs helping remapping keyboard values
i've stumbled uppon : SharpKeys

I'll test it :)

**Comment by seerge** [comment]
@lucidnx @Aurel100 I have to disappoint you but neither autohotkey nor sharpkeys won't work here. 

Main reason FN+Fx combinations on asus keyboards don't trigger a key press at all, they trigger a WMI event that only Asus Optimization service intercepts, and IF it was not intercepted  - then only G-Helper can catch (other apps can't)

The only way to implement proper FN lock is to disable Asus Optimization completely, and handle ALL hotkeys within G-helper. This means implement all actions - which is quite difficult :)) 

Therefore I don't plan to do it for now as it means rewriting whole asus optimization from scratch 😞

**Comment by Aurel100** [comment]
@seerge @lucidnx i've tried it anyway at as you said : Sharpkeys  doesn't recognize the key.
I understand that rewriting G-Helper would be a  nightmare so maybe I will let it go
(otherwise, maybe you could make  a new and different program that only works with Asus computers ? Not an expert but it would be worth to see how much ppl are in the same situation).

Anyway, G-helper is a great tool and i'm going to keep it running !
Thanks to you my computer can go to 9h on battery, vs 5 previously !
Huge thank you !

**Comment by lucidnx** [comment]
I think that you did not get it. There is no overwriting of GHelper, he will need to code whole functionality of ASUS Optilization service to implement hotkeys. It's too much work for almost no gain, maybe not possible at all...

**Comment by seerge** [comment]
@Aurel100 @lucidnx you can try a simple fn+lock in https://github.com/seerge/g-helper/releases/tag/v0.74

Keep in mind it works only in one way (i.e. Fx becomes Fn+Fx), doing opposite is unproportionally more complicated and invasive task
